### PR TITLE
fix(calcite-input): number inputs no longer display invalid numbers in localized input and pasting number values works correctly

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issues:** #2001 #2002 #2003

### CODE REVIEW

File: src/components/calcite-input/calcite-input.tsx [hunk 1]
```diff
@@ -495,12 +492,14 @@ export class CalciteInput {
...
     const previousValue = this.value;
-    this.value = this.type === "number" ? sanitizeDecimalString(value) : value;
-    if (this.shouldFormatNumberByLocale()) {
+    if (this.type === "number") {
+      this.value = sanitizeDecimalString(value);
       this.setLocalizedValue(value);
-    }
-    if (this.type === "number" && value.endsWith(".")) {
-      return;
+      if (value && value.endsWith(".")) {
```
Comment (driskull) [inline src/components/calcite-input/calcite-input.tsx:498]: Could this be simplified to `value?.endsWith(".")`?

File: src/components/calcite-input/calcite-input.tsx [hunk 2]
```diff
@@ -495,12 +492,14 @@ export class CalciteInput {
...
+        return;
+      }
+    } else {
```
Comment (driskull) [inline src/components/calcite-input/calcite-input.tsx:501]: I don't think this else is necessary if it returns above.

**Review by driskull — APPROVED** [review]
👍

**Comment by eriklharper** [comment]
Closing this in favor of separate PRs that tackle each issue.

**Pull request closed without merging**